.github/workflows/reusable_publish_image.yml

```diff
@@ -0,0 +1,246 @@
+# Reusable workflow for building and publishing container images with security best practices
+# This workflow implements a 4-stage secure build pipeline:
+# 1. Local build + 2. Secret & vulnerability scanning + 3. Push (Buildx emits SBOM & provenance) + 4. Signing & verification
+# Note: Repository secret scanning and Dockerfile linting are handled by reusable_ci.yml (MegaLinter)
+
+name: Build and Publish Image (Reusable)
+
+on:
+  workflow_call:
+    inputs:
+      allowed_identity_regex:
+        description: 'Regex for Cosign certificate identity (optional, defaults to org-level)'
+        required: false
+        type: string
+      component_name:
+        description: 'Component name (e.g., compass, beacon-distro)'
+        required: true
+        type: string
+      context_path:
+        description: 'Build context path (e.g., ./compass)'
+        required: true
+        type: string
+      dockerfile_path:
```

Comment:
Could we call `containerfile` instead of `dockerfile` to be more technology-agnostic?